src/Data/Argonaut/Decode.purs

```diff
@@ -37,6 +44,44 @@ import qualified Data.Map as Map
 class DecodeJson a where
   decodeJson :: Json -> Either String a
 
+-- | Decode `Json` representation of a value which has a `Generic` type.
+gDecodeJson :: forall a. (Generic a) => Json -> Either String a
+gDecodeJson = map (fromJust <<< fromSpine)
```

Comment:
We should eliminate this `fromJust` and instead return a `Left` with some appropriate error message.